app/tests/pages/analyst/application/[applicationId].test.tsx

```diff
@@ -0,0 +1,697 @@
+import { screen, within } from '@testing-library/react';
+import mockFormData from 'tests/utils/mockFormData';
+import { acknowledgementsEnum } from 'formSchema/pages/acknowledgements';
+import PageTestingHelper from '../../../utils/pageTestingHelper';
+import Application from '../../../../pages/analyst/application/[applicationId]';
+import compiledApplicationIdQuery, {
+  ApplicationIdQuery,
+} from '../../../../__generated__/ApplicationIdQuery.graphql';
+
+const mockQueryPayload = {
+  Query() {
+    return {
+      applicationByRowId: {
+        formData: {
+          jsonData: mockFormData,
+        },
+      },
+      session: {
+        sub: '4e0ac88c-bf05-49ac-948f-7fd53c7a9fd6',
+      },
+    };
+  },
+};
+
+const mockEmptyFormDataPayload = {
+  Query() {
+    return {
+      applicationByRowId: {
+        formData: {
+          jsonData: {},
+        },
+      },
+      session: {
+        sub: '4e0ac88c-bf05-49ac-948f-7fd53c7a9fd6',
+      },
+    };
+  },
+};
+
+const pageTestingHelper = new PageTestingHelper<ApplicationIdQuery>({
+  pageComponent: Application,
+  compiledQuery: compiledApplicationIdQuery,
+  defaultQueryResolver: mockQueryPayload,
+  defaultQueryVariables: {
+    rowId: 1,
+  },
+});
+
+describe('The analyst view application page', () => {
+  beforeEach(() => {
+    pageTestingHelper.reinit();
+    pageTestingHelper.setMockRouterValues({
+      query: { applicationId: '1' },
+    });
+  });
+
+  it('displays the correct nav links when user is logged in', () => {
+    pageTestingHelper.loadQuery();
+    pageTestingHelper.renderPage();
+
+    expect(screen.getByText('Logout')).toBeInTheDocument();
+  });
+
+  it('displays the correct title', () => {
+    pageTestingHelper.loadQuery();
+    pageTestingHelper.renderPage();
+
+    expect(
+      screen.getByRole('heading', { name: 'Application' })
+    ).toBeInTheDocument();
+  });
+
+  it('should have correct heading styles', () => {
```

Comment:
Looks like there's a lot of duplication (about 450 lines of test code?) with the test on the applicant's side. Any way we can refactor these tests so that we avoid the duplication and are consistent in what we test for both pages?

Reply:
Okay I've made all of the requested changes except for this (I could also add a happo screenshot once that PR gets merged). It might require a bit of work as this test uses `pageTestingHelper` while `ApplicationForm` tests use `componentTestingHelper`.

Happy to discuss this after standup about how to go about reusing some of the tests.

Reply:
@matthieu-foucault I moved shared tests to `/tests/components/ReviewTheme.ts` which is an exported function. Let me know what you think of it.